### PR TITLE
New doc: Identify and Kill Queries with MySQL Command-Line Tool

### LINKED ISSUE #634: Document how to identify and kill queries using the mysql-cli
In the event a flood of slow transactions hits the database, queries can be killed from the mysql-cli using the `kill` command. 

http://smallbusiness.chron.com/kill-long-running-queries-mysql-48524.html

### PR DESCRIPTION
**Pull request by rachelwhitton**
Closes #634 

New doc: **Identify and Kill Queries with MySQL Command-Line Tool**

@timani can you review?

### CODE REVIEW

File: source/docs/articles/sites/database/kill-mysql-queries.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+<div class="alert alert-info" role="alert">
+<h4>Note</h4>
+Replace <code>&lt;thread_id&gt;</code> with the ID of the query you wish to terminate.
+</div>
+## Kill All Queries
+If a large number of bad requests are blocking valid queries you may want to clear them out without having to run `kill` on every individual thread.
+
+Use the following command to fetch information from the `PROCESSLIST` table and generate an external file with `kill` commands:
+```
+mysql> select concat('KILL ',id,';') from information_schema.processlist where user='root' into outfile '/tmp/killallthreads.txt';
```
Comment (rachelwhitton) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: @timani or @ari-gold Would `root` be the correct user when trying to kill all queries? Any thoughts on unwanted results, such as killing necessary platform related queries?

**Comment by rachelwhitton** [comment]
@bmackinney can you review - what docs should link to this one?

File: source/docs/articles/sites/database/kill-mysql-queries.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+<h4>Note</h4>
+Replace <code>&lt;thread_id&gt;</code> with the ID of the query you want to terminate.
+</div>
+
+## Kill All Queries
+If a large number of bad requests are blocking valid queries, you can clear them out without having to run `kill` on every individual thread.
+
+Use the following command to fetch information from the `PROCESSLIST` table and generate an external file with `kill` commands:
+```
+mysql> select concat('KILL ',id,';') from information_schema.processlist where user='root' into outfile '/tmp/killallthreads.txt';
```
Comment (rachelwhitton) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: The user should be 'pantheon' 

We may need to remove this section due to using the `/tmp` dir. The txt file is successfully created, but using `source /tmp/file.txt` returns the following: 
```
Failed to open file 'killallthreads.txt', error: 2
```
@ari-gold any tips?

Comment (ari-gold) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: @rachelwhitton Found another option, just tested and seems to work fine. This generates a query that you copy and paste to kill all processes:

```
SELECT GROUP_CONCAT(CONCAT('KILL ',id,';') SEPARATOR ' ') 'Paste the following query to kill all processes' FROM information_schema.processlist WHERE user<>'system user'\
```
adapted from http://dba.stackexchange.com/questions/2634/kill-all-queries-mysql

kill-all-queries

File: source/docs/articles/sites/database/kill-mysql-queries.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+title: Identify and Kill Queries with MySQL Command-Line Tool
+description: Learn how to identify and kill long-running MySQL queries on your WordPress or Drupal site in a few commands.
+keywords: mysql, mysql query, mysql queries, kill query, kill queries, kill mysql, kill mysql queries, queries, query, debug mysql, debug, mysql error
+---
+Long-running MySQL queries keep other transactions from accessing the necessary tables to execute a request, leaving your users on hold. To kill these queries, all you need is a local MySQL client such as [MySQL Workbench](http://dev.mysql.com/downloads/tools/workbench/), [Sequel Pro](http://www.sequelpro.com/download), or [Navicat](http://www.navicat.com/download).
```
Comment (ari-gold) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: Suggest changing 2nd sentence to "To kill these queries, you'll need to [Access the appropriate environment's MySQL Database](/docs/articles/local/accessing-mysql-databases).

File: source/docs/articles/sites/database/kill-mysql-queries.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+title: Identify and Kill Queries with MySQL Command-Line Tool
+description: Learn how to identify and kill long-running MySQL queries on your WordPress or Drupal site in a few commands.
+keywords: mysql, mysql query, mysql queries, kill query, kill queries, kill mysql, kill mysql queries, queries, query, debug mysql, debug, mysql error
+---
+Long-running MySQL queries keep other transactions from accessing the necessary tables to execute a request, leaving your users on hold. To kill these queries, all you need is a local MySQL client such as [MySQL Workbench](http://dev.mysql.com/downloads/tools/workbench/), [Sequel Pro](http://www.sequelpro.com/download), or [Navicat](http://www.navicat.com/download).
+
+## Create a New MySQL Connection
```
Comment (ari-gold) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: Suggest removing this section. They can read  "Accessing MySQL database" if they don't know how.

File: source/docs/articles/sites/database/kill-mysql-queries.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+title: Identify and Kill Queries with MySQL Command-Line Tool
+description: Learn how to identify and kill long-running MySQL queries on your WordPress or Drupal site in a few commands.
+keywords: mysql, mysql query, mysql queries, kill query, kill queries, kill mysql, kill mysql queries, queries, query, debug mysql, debug, mysql error
+---
+Long-running MySQL queries keep other transactions from accessing the necessary tables to execute a request, leaving your users on hold. To kill these queries, all you need is a local MySQL client such as [MySQL Workbench](http://dev.mysql.com/downloads/tools/workbench/), [Sequel Pro](http://www.sequelpro.com/download), or [Navicat](http://www.navicat.com/download).
+
+## Create a New MySQL Connection
+Each site environment (Multidev, Dev, Test and Live) has a separate database. Click **Connection Info** in the Dashboard of the environment you want to investigate. Paste the provided database command into your terminal to establish a connection.
+
+### Create a Secure Connection
```
Comment (ari-gold) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: Suggest removing this section

File: source/docs/articles/sites/database/kill-mysql-queries.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+You can create a secure connection using SSH tunnels, but it is not required. For more information, see [SSH Tunnels for Secure Connections to Pantheon Services](/docs/articles/local/ssh-tunnels-for-secure-connections-to-pantheon-services).
+<div class="alert alert-info" role="alert">
+<h4>Note</h4>
```
Comment (ari-gold) [inline source/docs/articles/sites/database/kill-mysql-queries.md]: Suggest removing this note

**Comment by ari-gold** [comment]
+1 :hocho:

**Comment by rachelwhitton** [comment]
Thanks for the help @ari-gold - your suggestion worked perfectly!

**Pull request merged**